scripts/s3manage.py

```diff
@@ -0,0 +1,568 @@
+
+        s3client.copy_object(
+            Bucket=BUCKET_NAME,
+            CopySource=BUCKET_NAME +
+            '/' +
+            src_key_name,
+            Key=n +
+            '.html',
+            ACL='public-read')
+    print "Special files"
+    appcache = None
+    files = list(bucket.objects.filter(Prefix='{}/geoadmin.'.format(version)).all())
+    if len(files) > 0:
+        appcache = os.path.basename(sorted(files)[-1].key)
+    for j in ('robots.txt', 'checker', appcache):
+        src_key_name = '{}/{}'.format(version, j)
+        print src_key_name, os.path.basename(src_key_name)
+        try:
+            s3client.copy_object(
+                Bucket=BUCKET_NAME,
+                CopySource=BUCKET_NAME + '/' + src_key_name,
+                Key=os.path.basename(src_key_name),
+                CopySourceIfModifiedSince=datetime(2015, 1, 1),
+                ACL='public-read')
+        except botocore.exceptions.ClientError as e:
+            print "Cannot copy {}: {}".format(j, e)
+
+    print("\n\nPlease check it on {}".format(get_url()))
+    print("  and {}".format(get_url('src/index.html')))
+
+
+def get_url(key_name='index.html'):
+    bucket_location = BUCKET_LOCATION
+    object_url = "https://s3-{0}.amazonaws.com/{1}/{2}".format(
+        bucket_location,
+        BUCKET_NAME,
+        key_name)
+    return object_url
+
+
+def main():
+    global BUCKET_NAME
+
+    if BUCKET_NAME is None:
+        print "Please define the BUCKET_NAME you want to deploy."
+        sys.exit(2)
+    if s3client is None:
+        init_connection()
+
+    if len(sys.argv) < 2:
+        usage()
+        sys.exit()
+
+    if str(sys.argv[1]) == 'upload':
+
+        if len(sys.argv) < 4:
+            usage()
+            sys.exit()
+        target = sys.argv[2]
+        if target not in ['dev', 'int', 'prod']:
+            usage()
+            sys.exit()
+
+        BUCKET_NAME = os.environ.get("S3_MF_GEOADMIN3_{}".format(target.upper()))
+
+        if BUCKET_NAME is None:
+            print "Please define the BUCKET_NAME you want to deploy."
+            sys.exit(2)
+
+        SNAPSHOT = os.environ.get('SNAPSHOT')
+        if SNAPSHOT is not None:
+            try:
+                SNAPSHOT = int(SNAPSHOT)
+            except ValueError:
+                print "SNAPSHOT name is not in the form 'YYYYMMDDhhmm'"
+                sys.exit(2)
+
+            base_dir = os.path.join(BASEDIR, 'snapshots', str(SNAPSHOT), 'geoadmin/code/geoadmin')
+
+            if not os.path.isdir(base_dir):
+                print "No code found in directory {}".format(base_dir)
+                sys.exit(2)
+            # VERSION = SNAPSHOT
+        else:
+            if len(sys.argv) == 4:
+                base_dir = os.path.abspath(sys.argv[3])
+                if not os.path.isdir(base_dir):
+                    print "No code found in directory {}".format(base_dir)
+                    sys.exit(2)
+            else:
+                base_dir = os.getcwd()
```

Comment:
This is in contradiction with line 484 and is dead code as you will exit if the `len(sys.argv) < 4`